modules/tts/tts-adapters/tts-adapter-aws/src/main/java/org/daisy/pipeline/tts/aws/impl/AWSPollyTTSEngine.java

```diff
@@ -0,0 +1,207 @@
+import software.amazon.awssdk.services.polly.model.OutputFormat;
+
+/**
+ * TTS adapter for Amazon Polly
+ * @author mmartida - original author
+ * @author Nicolas Pavie - updates and refactoring for DAISY Pipeline 2, version 1.15+
+ */
+public class AWSPollyTTSEngine extends TTSEngine {
+
+	private final static URL SSML_TRANSFORMER = URLs.getResourceFromJAR("/transform-ssml.xsl", AWSPollyTTSEngine.class);
+
+	private final static Logger LOGGER = LoggerFactory.getLogger(AWSPollyTTSEngine.class);
+
+	private static final AudioFormat AF_PCM_POLLY = new AudioFormat(AudioFormat.Encoding.PCM_SIGNED, 16000f, 16, 1, 2, 16000f, false);
+
+	/**
+	 * Client object provided by AWS SDK to interact with the polly engines
+	 */
+	private final PollyClient client;
+
+	/**
+	 * Map of AWS voices available to the user for each engine
+	 */
+	private final HashMap<Engine, DescribeVoicesResponse> awsVoicesDescription = new HashMap<>();
+
+	private int mPriority;
+
+	/**
+	 * Constructor for the AWSPollyTTSEngine
+	 *
+	 * @param service the bound service
+	 * @param accessKey the AWS account access key
+	 * @param secretKey the AWS account secret key
+	 * @param region the AWS region
+	 * @throws SynthesisException
+	 */
+	public AWSPollyTTSEngine(AWSPollyTTSService service, String accessKey, String secretKey, String region, int priority)
+			throws SynthesisException {
+
+		super(service);
+		this.client = PollyClient.builder()
+			.credentialsProvider(() -> new AwsCredentials() {
+				@Override
+				public String accessKeyId() {
+					return accessKey;
+				}
+
+				@Override
+				public String secretAccessKey() {
+					return secretKey;
+				}
+			})
+			.region(Region.of(region))
+			.build();
+		this.mPriority = priority;
+	}
+
+	/**
+	 * Get the list of available voices
+	 *
+	 * @throws SynthesisException
+	 * @throws InterruptedException
+	 */
+	@Override
+	public Collection<Voice> getAvailableVoices() throws SynthesisException, InterruptedException {
+		synchronized (this.client){
+			// Order of priorities based on
+			Engine[] engines = new Engine[] {
+				Engine.STANDARD,
+				Engine.NEURAL,
+				Engine.GENERATIVE,
+				Engine.LONG_FORM,
+			};
+			Collection<Voice> voices = new ArrayList<>();
+			if (awsVoicesDescription.isEmpty()) {
+				// fetch all voices for all engines
+				for (Engine _engine : engines) {
+					try {
+						DescribeVoicesResponse awsvoices = this.client.describeVoices(b -> b.engine(_engine));
+						awsVoicesDescription.put(_engine, awsvoices);
+					} catch (Exception e) {
+						LOGGER.error("Error fetching AWS voices for engine {}: {}", _engine, e.getMessage());
+					}
+				}
+			}
+			for (Engine engine : engines) {
+				if(awsVoicesDescription.containsKey(engine)) {
+					voices.addAll(
+						awsVoicesDescription.get(engine).voices()
+							.stream()
+							.map(i -> new Voice("aws", i.idAsString() + " (" + engine.toString() + ")",
+							                    new Locale(i.languageName()),
```

Comment:
In other places locales are parsed like this:

```java
(new Locale.Builder()).setLanguageTag(info.getLocale().replace("_", "-")).build()
```

In the original code there was some language normalisation going on.

- `arb` was replaced by `ar` with the same meaning.
- `cmn-CN` was replaced by `zh_CN`

(Originally this transformation was done in `VoiceInfo.tagToLocale)(`, not in `TTSEngine.getAvailableVoices()`, but the result should be the same.)

I wonder whether this normalisation is still needed. Note that we have the `org.daisy.pipeline.common.NormalizeLang` class (in common-utils) which could _in theory_ be used for it.